servers/quarkus-cli/src/main/java/org/projectnessie/quarkus/cli/ImportRepository.java

```diff
@@ -0,0 +1,237 @@
+          spec.commandLine()
+              .getErr()
+              .println(
+                  "The Nessie repository already exists and is not empty, aborting. "
+                      + "Provide the "
+                      + ERASE_BEFORE_IMPORT
+                      + " option if you want to erase the repository.");
+          return 100;
+        }
+
+        erase = true;
+      }
+    }
+
+    // Always perform erase + initialize to reset any non-obvious contents (global log, global
+    // state, ref-log, repository description, etc).
+    out.println("Erasing potentially existing repository...");
+    databaseAdapter.eraseRepo();
+    databaseAdapter.initializeRepo("main");
+    try {
+      databaseAdapter.delete(BranchName.of("main"), Optional.empty());
+    } catch (ReferenceNotFoundException | ReferenceConflictException e) {
+      throw new RuntimeException(e);
+    }
+
+    ImportResult importResult =
+        builder.progressListener(new ImportProgressListener(out)).build().importNessieRepository();
+
+    out.printf(
+        "Imported Nessie repository, %d commits, %d named references.%n",
+        importResult.importedCommitCount(), importResult.importedReferenceCount());
+
+    if (!noOptimize) {
+      out.println("Optimizing...");
+
+      CommitLogOptimization.builder()
+          .headsAndForks(importResult.headsAndForkPoints())
+          .databaseAdapter(databaseAdapter)
+          .build()
+          .optimize();
+
+      out.println("Finished commit log optimization.");
+    }
+
+    return 0;
+  }
+
+  /** Mostly paints dots - but also some numbers about the progress. */
+  private static final class ImportProgressListener implements ProgressListener {
+
+    private final PrintWriter out;
+    private int count;
+    private boolean dot;
+    private ExportMeta exportMeta;
+
+    public ImportProgressListener(PrintWriter out) {
+      this.out = out;
+    }
+
+    @Override
+    public void progress(@Nonnull ProgressEvent progress, ExportMeta meta) {
+      switch (progress) {
+        case END_META:
+          this.exportMeta = meta;
+          break;
+        case START_COMMITS:
+          out.printf("Importing %d commits...%n", exportMeta.getCommitCount());
+          count = 0;
+          dot = false;
+          break;
+        case END_COMMITS:
+          if (dot) {
+            out.println();
+          }
+          out.printf("%d commits imported.%n%n", count);
+          break;
+        case START_NAMED_REFERENCES:
+          out.printf("Importing %d named references...%n", exportMeta.getNamedReferencesCount());
+          count = 0;
+          dot = false;
+          break;
+        case COMMIT_WRITTEN:
+        case NAMED_REFERENCE_WRITTEN:
+          count++;
+          if ((count % 10) == 0) {
+            out.print('.');
+            dot = true;
+          }
+          if ((count % 500) == 0) {
+            out.printf(" %d%n", count);
+            dot = false;
+          }
```

Comment:
nit: may be move to a base class, shared with `ExportProgressListener`?

Reply:
Yea - I know, it's a couple duplicated code. Not sure whether sharing would make it cleaner though - abstraction + overrides for the differences probably doesn't let it look better.

Reply:
fair enough